src/test/java/com/microsoft/sqlserver/jdbc/connection/TimeoutTest.java

```diff
@@ -129,134 +139,172 @@ public void testDMLoginTimeoutNotApplied() {
                     .getConnection("jdbc:sqlserver://" + randomServer + ";loginTimeout=" + timeout)) {
                 fail(TestResource.getResource("R_shouldNotConnect"));
             } catch (Exception e) {
-                timerEnd = System.currentTimeMillis();
+                totalTime = System.currentTimeMillis() - timerStart;
 
                 assertTrue(
-                        (e.getMessage().contains(TestResource.getResource("R_tcpipConnectionToHost")))
-                                || ((isSqlAzure() || isSqlAzureDW())
-                                                                     ? e.getMessage()
-                                                                             .contains(TestResource
-                                                                                     .getResource("R_connectTimedOut"))
-                                                                     : false),
+                        (e.getMessage().toLowerCase()
+                                .contains(TestResource.getResource("R_tcpipConnectionToHost").toLowerCase()))
+                                || ((isSqlAzure() || isSqlAzureDW()) ? e.getMessage().toLowerCase()
+                                        .contains(TestResource.getResource("R_connectTimedOut").toLowerCase()) : false),
                         e.getMessage());
             }
-            verifyTimeout(timerEnd - timerStart, timeout);
+
+            // time should be < connection string loginTimeout
+            assertTrue(totalTime < TimeUnit.SECONDS.toMillis(timeout),
+                    "total time: " + totalTime + " loginTimeout: " + TimeUnit.SECONDS.toMillis(timeout));
         } finally {
             DriverManager.setLoginTimeout(0); // Default to 0 again
         }
     }
 
+    // Test connect retry set to 0 (disabled)
+    @Test
+    public void testConnectRetryDisable() {
+        long totalTime = 0;
+        long timerStart = System.currentTimeMillis();
+        int interval = defaultTimeout; // long interval so we can tell if there was a retry
+        long timeout = defaultTimeout * 2; // long loginTimeout to accommodate the long interval
+
+        // non existent server with long loginTimeout, should return fast if no retries at all
+        try (Connection con = PrepUtil.getConnection(
+                "jdbc:sqlserver://" + randomServer + ";transparentNetworkIPResolution=false;loginTimeout=" + timeout
+                        + ";connectRetryCount=0;connectInterval=" + interval)) {
+            fail(TestResource.getResource("R_shouldNotConnect"));
+        } catch (Exception e) {
+            totalTime = System.currentTimeMillis() - timerStart;
```

Comment:
Not sure if it will be flaky. But, just another opinion, can we programmatically check logs for the retries instead (same for some of the timer tests below)? Maybe easier said than done.